pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -315,11 +316,14 @@ public void indexRow(GenericRow row)
 
       FieldSpec fieldSpec = _schema.getFieldSpecFor(columnName);
       SegmentDictionaryCreator dictionaryCreator = _dictionaryCreatorMap.get(columnName);
-
-      if (fieldSpec.isSingleValueField()) {
-        indexSingleValueRow(dictionaryCreator, columnValueToIndex, creatorsByIndex);
-      } else {
-        indexMultiValueRow(dictionaryCreator, (Object[]) columnValueToIndex, creatorsByIndex);
+      try {
+        if (fieldSpec.isSingleValueField()) {
+          indexSingleValueRow(dictionaryCreator, columnValueToIndex, creatorsByIndex);
+        } else {
+          indexMultiValueRow(dictionaryCreator, (Object[]) columnValueToIndex, creatorsByIndex);
+        }
+      } catch (JsonParseException jpe) {
```

Comment:
Thanks ! This will be pretty useful for debugging.

Reply:
Does it make sense to catch this exception to where json index is created (JsonIndexCreator?) or where parsing actually happens.

Reply:
In any level higher than `SegmentColumnarIndexCreator`, it is not easy to figure out the column name. IIUC, callers of this function, are indexing a row and there maybe different indices built for different columns. This is the first function that is iterating through all the columns and building an index for each one. So columnName is obvious. 

This function passes the column index value instead of names to `IndexSingleValueRow`. So names are not available again.

Reply:
That makes sense, this seems like the best place to put the exception without doing some major refactoring.